todo_app/tests/integration/test_app_routes.py

```diff
@@ -0,0 +1,135 @@
+        return StubResponse(fake_response_data={**kwargs['data'], 'id': 'test-list-new'})
+
+    # List endpoints
+    for test_list in [test_to_do_list, test_doing_list, test_done_list]:
+        if method == "GET" and url == f"{trello_api_base_url}/lists/{test_list['id']}":
+            return StubResponse(fake_response_data=test_list)
+        if method == "GET" and url == f"{trello_api_base_url}/lists/{test_list['id']}/cards":
+            return StubResponse(fake_response_data=test_list['cards'])
+    if method == "GET" and url == f"{trello_api_base_url}/lists/test-list-new/cards":
+        return StubResponse(fake_response_data=[])
+
+    # Card endpoints
+    if method == "POST" and url == f"{trello_api_base_url}/cards" \
+            and all(key in kwargs['data'] for key in ['name', 'desc', 'idList']):
+        return StubResponse(fake_response_data={**kwargs['data'], 'id': 'test-card-new'})
+    for test_card in [test_to_do_card, test_doing_card, test_done_card]:
+        if method == "GET" and url == f"{trello_api_base_url}/cards/{test_card['id']}":
+            return StubResponse(fake_response_data=test_card)
+        if method == "PUT" and url == f"{trello_api_base_url}/cards/{test_card['id']}" \
+                and any(key in kwargs['data'] for key in ['name', 'desc', 'idList']):
+            return StubResponse(fake_response_data={**test_card, **kwargs['data']})
+        if method == "DELETE" and url == f"{trello_api_base_url}/cards/{test_card['id']}":
+            return StubResponse()
+
+
+def test_index(monkeypatch, client):
+    monkeypatch.setattr(requests, 'request', request_stub)
+
+    response = client.get("/")
+    assert response.status_code == 200
+
+    page_content = response.data.decode()
+    assert test_to_do_card['name'] in page_content
+    assert test_to_do_card['desc'] in page_content
+    assert test_doing_card['name'] in page_content
+    assert test_doing_card['desc'] in page_content
+    assert test_done_card['name'] in page_content
+    assert test_done_card['desc'] in page_content
+
+
+def test_create_item(monkeypatch, client):
+    monkeypatch.setattr(requests, 'request', request_stub)
+
+    response = client.post("/create", data={'title': 'New task'})
+    assert response.status_code == 302
+
+
+def test_update_item(monkeypatch, client):
+    monkeypatch.setattr(requests, 'request', request_stub)
+
+    response = client.post(f"/update/{test_to_do_card['id']}", data={'title': 'Altered task', 'desc': 'Altered description'})
+    assert response.status_code == 302
+
+    response = client.post(f"/update/{test_to_do_card['id']}", data={'idList': test_doing_list['id']})
+    assert response.status_code == 302
+
+
+def test_delete_item(monkeypatch, client):
+    monkeypatch.setattr(requests, 'request', request_stub)
+
+    response = client.post(f"/delete/{test_to_do_card['id']}")
+    assert response.status_code == 302
```

Comment:
These tests currently fail because (as I understand it!) your ItemIdConverter insists that item ids match a 24 length alphanumeric, but your test ids aren't of that form!

Reply:
Whoops, forgot to rerun the tests after merging main! Not ideal